lightning/restore/restore.go

```diff
@@ -1057,23 +1055,26 @@ func (t *TableRestore) restoreEngine(
 	indexEngine *kv.OpenedEngine,
 	engineID int32,
 	cp *EngineCheckpoint,
-) (*kv.ClosedEngine, *worker.Worker, error) {
+) (*kv.ClosedEngine, error) {
 	if cp.Status >= CheckpointStatusClosed {
-		w := rc.closedEngineLimit.Apply()
 		closedEngine, err := rc.backend.UnsafeCloseEngine(ctx, t.tableName, engineID)
 		// If any error occurred, recycle worker immediately
 		if err != nil {
-			rc.closedEngineLimit.Recycle(w)
```

Comment:
the `closedEngineLimit` was introduced in #119 to put a "back-pressure" if Importer is slower than Lightning. i'm not sure if we should remove the worker entirely.

> closed-engine: 一个 Lightning batch 都对应了一个 Importer 的 Engine，如果一个 batch 完全写入到 Importer，就会关闭这个 Engine，直到这个 Engine Import 到 TiKV 之后才会对这个 Engine 进行清理，如果有 closed engine 的 Idle 数量少，表示 Lightning 的速度比 Importer 的速度快，等 closed engine 的 Idle 数量完全为 0 时，Lightning 会暂停写入。

Reply:
Thanks. I think we can import KV async and do not block this goroutine so that it can import other table. I have talked this with @glorv . But I will revert it in this PR in order not to introduce irrelevant changes.

Reply:
From the current implementation, `closedEngineLimit` mainly affect on how many engines than can be imported in parallel. This pr just move the apply action from close engine to before start import. Does this really matter 🤔？ @kennytm

Reply:
if they have the same effect then LGTM on this part.